### PR TITLE
Display ordered items and total price to confirm before checkout

### PR DESCRIPTION
**Pull request by marizaswart**
### Description
This step will show the final order of the visitor to confirm the products selected. It will confirm the products, product prices, delivery fee and total price. From this page the visitor is able to click pay with card and continue to the stripe payment to finalise the order.

### What we have learnt
* How to create and order so a user can select products and display these products on the checkout page
* How to display the different quantities of the orders and interaction with different restaurants on one order
### Screenshot
screen shot 2018-01-19 at 13 01 48

### CODE REVIEW

File: features/step_definitions/basic_steps.rb [hunk 1]
```diff
@@ -46,3 +37,18 @@ def page_path_from(page_name)
...
+def page_path_from(page_name)
+  case page_name.downcase
+    when 'thaitanic' then
+      restaurant_path(Restaurant.find_by(name: page_name))
+    when 'landing' then
+      root_path
+    when 'sign up' then
+      new_user_registration_path
+    when 'cart' then
+      charges_path()
```
Comment (ScottyyyB) [inline features/step_definitions/basic_steps.rb]: Small thing, but the parenthesis after line 50 is not needed I'm pretty sure, other than that everything looks good!

Comment (aljoh) [reply features/step_definitions/basic_steps.rb]: Fixed :)

**Review by zuzanna81 — COMMENTED** [review]
In general would like to have a walkthrough the implementation of this feature.

Otherwise looks good

**Review by diraulo — COMMENTED** [review]
Update this with latest code from upstream and i'll take another look

**Comment by diraulo** [comment]
If you have screenshots to add to the description, please do so, otherwise remove the title altogether. Same goes for `what we have learnt` section of your description.

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -46,3 +37,18 @@ def page_path_from(page_name)
 Then("I should not see {string}") do |content|
   expect(page).not_to have_content content
 end
+
+def page_path_from(page_name)
+  case page_name.downcase
+    when 'thaitanic' then
```
Comment (diraulo) [inline features/step_definitions/basic_steps.rb]: If the action is on the next line the `then` isn't required. Also you need a default case. What happens when we don't match any of the pre-defined page names?

**Review by diraulo — DISMISSED** [review]
🚫

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -18,6 +18,10 @@ def update
     end
   end
 
+  def show
+    @order = Order.find(session[:order_id])
```
Comment (tochman) [inline app/controllers/orders_controller.rb]: Why use session and not params to find the order?

File: app/views/partials/_navbar.html.erb [hunk 4]
```diff
@@ -23,7 +23,9 @@
         </ul>
         <ul class="navbar-nav ml-auto">
             <li class="nav-item">
-                <a class="glyphicon glyphicon-shopping-cart" href="" id=order-icon> Cart</a>
+              <% if session[:order_id] %>
+                <%= link_to ' Checkout', order_path(session[:order_id]), class: 'glyphicon glyphicon-shopping-cart', id: "order-icon" %>
```
Comment (tochman) [inline app/views/partials/_navbar.html.erb]: Here you are sending the order as params so please use that in your controller

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/views/partials/_navbar.html.erb (hunk 4 above)
Comment (aljoh) [reply app/views/partials/_navbar.html.erb]: Yes, we'll fix that

Comment (aljoh) [reply app/views/partials/_navbar.html.erb]: Fixed

Comment (tochman) [reply app/views/partials/_navbar.html.erb]: Why the space before `' Checkout'` ?

File: app/controllers/orders_controller.rb (hunk 3 above)
Comment (aljoh) [reply app/controllers/orders_controller.rb]: Fixed

File: app/views/partials/_navbar.html.erb (hunk 4 above)
Comment (aljoh) [reply app/views/partials/_navbar.html.erb]: Fixed it. Apparently was used as a "duct tape" solution to put a space between the checkout icon and the text. We'll do it with css instead

**Review by tochman — APPROVED** [review]

**Review by tochman — APPROVED** [review]
Good job

**Pull request merged**